src/com/goide/inspections/GoUsedAsValueInCondition.java

```diff
@@ -0,0 +1,122 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.goide.inspections;
+
+import com.goide.GoConstants;
+import com.goide.psi.*;
+import com.goide.psi.impl.GoElementFactory;
+import com.intellij.codeInspection.LocalInspectionToolSession;
+import com.intellij.codeInspection.LocalQuickFixOnPsiElement;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.text.StringUtil;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.intellij.util.NotNullFunction;
+import org.jetbrains.annotations.Nls;
+import org.jetbrains.annotations.NotNull;
+
+import static com.intellij.codeInspection.ProblemHighlightType.GENERIC_ERROR_OR_WARNING;
+
+public class GoUsedAsValueInCondition extends GoInspectionBase {
+  public static final String QUICK_FIX_NAME = "Convert to '==''";
+
+  private static final NotNullFunction<PsiElement, String> GET_TEXT_FUNCTION = new NotNullFunction<PsiElement, String>() {
+    @NotNull
+    @Override
+    public String fun(@NotNull PsiElement element) {
+      return element.getText();
+    }
+  };
+
+  @NotNull
+  @Override
+  protected GoVisitor buildGoVisitor(@NotNull final ProblemsHolder holder, @NotNull LocalInspectionToolSession session) {
+    return new GoVisitor() {
+
+      @Override
+      public void visitAssignmentStatement(@NotNull GoAssignmentStatement o) {
+        if (o.getParent() != null &&
+            o.getParent() instanceof GoIfStatement &&
+            ((GoIfStatement)o.getParent()).getExpression() == null) {
+          String left = StringUtil.join(o.getLeftHandExprList().getExpressionList(), GET_TEXT_FUNCTION, ", ");
+          String right = StringUtil.join(o.getExpressionList(), GET_TEXT_FUNCTION, ", ");
+
+          holder.registerProblem(o, left + " = " + right + " used as value", GENERIC_ERROR_OR_WARNING, new GoAssignmentToComparisonQuickFix(o));
+        }
+      }
+      @Override
+      public void visitShortVarDeclaration(@NotNull GoShortVarDeclaration o) {
+        if (o.getParent() != null &&
+            o.getParent().getParent() != null &&
+            o.getParent().getParent() instanceof GoIfStatement &&
+            ((GoIfStatement)o.getParent().getParent()).getExpression() == null) {
+
+          String left = StringUtil.join(o.getVarDefinitionList(), GET_TEXT_FUNCTION, ", ");
+          String right = StringUtil.join(o.getExpressionList(), GET_TEXT_FUNCTION, ", ");
+
+          holder.registerProblem(o, left + " := " + right + " used as value", GENERIC_ERROR_OR_WARNING, new GoAssignmentToComparisonQuickFix(o));
+        }
+      }
+    };
+  }
+
+  public static class GoAssignmentToComparisonQuickFix extends LocalQuickFixOnPsiElement {
```

Comment:
You don't need LocalQuickFixOnPsiElement because this fix is used in inspection only and it the same element in inspection report and in quick with. Please reimplement with LocalQuickFix.

Also I think it shouldn't be public